operator/pkg/util/cert/util.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cert
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/istio/security/pkg/cmd"
+	k8ssecret "istio.io/istio/security/pkg/k8s/secret"
+	"istio.io/istio/security/pkg/pki/util"
+	"istio.io/pkg/log"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// CASecret defines plugged-in secret name
+	PlugInSecret = "cacerts"
+	// RootCertName defines plugged-in root cert file name
+	RootCertName = "root-cert.pem"
+	// CAKeyName defines plugged-in ca private key file name
+	CAKeyName = "ca-key.pem"
+	// CACertName  defines plugged-in ca cert file name
+	CACertName = "ca-cert.pem"
+	//CertChainName  defines plugged-in cert-chain file name
+	CertChainName = "cert-chain.pem"
+
+	caKeySize = 2048
+)
+
+func PlugInProvisionedCert(client kubernetes.Interface, certDir, ns string, dryRun bool) error {
```

Comment:
Why in operator ? This seems like a general purpose util, should be in top level pkg/ 
( not even istioctl specific )

I would imagine unit tests and running Istio without k8s should start using this code - many unit tests don't run in a k8s env and so fail to check TLS mode or use very different kind of certs than prod. 

@nmittler

Reply:
moved to istio/pkg